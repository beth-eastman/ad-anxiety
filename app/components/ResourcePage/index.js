/**
 * Contains a list of different resource pages (Links, Books, Articles, and Facts).
 *
 * Bethany Eastman
 */
import React, { PropTypes } from 'react';
import { Card, List, ListItem } from 'material-ui';

export default class ResourcePage extends React.Component {

  /* Change AppBar title */
  componentWillMount() {
    this.props.setPageTitle("Resources"); // eslint-disable-line
    this.props.changeIcon('Home'); // eslint-disable-line
  }

  /* Create a list of resources that directs the user to the corresponding content
   * page when selected */
  render() {
    return (
      <Card>
        <List>
          <ListItem
            primaryText="Links"
            onTouchTap={() => {
              this.props.router.push('/resources/links');
            }}
          />
          <ListItem
            primaryText="Books"
            onTouchTap={() => {
              this.props.router.push('/resources/books');
            }}
          />
          <ListItem
            primaryText="Articles"
            onTouchTap={() => {
              this.props.router.push('/resources/articles');
            }}
          />
          <ListItem
            primaryText="Facts"
            nestedItems={[
              <ListItem
                key={0}
                primaryText="Understanding Anxiety"
                onTouchTap={() => {
                  this.props.router.push('/resources/facts/0');
                }}
              />,
              <ListItem
                key={1}
                primaryText="Managing Anxiety and Worry"
                onTouchTap={() => {
                  this.props.router.push('/resources/facts/1');
                }}
              />,
              <ListItem
                key={2}
                primaryText="Causes of Anxiety"
                onTouchTap={() => {
                  this.props.router.push('/resources/facts/2');
                }}
              />,
            ]}
          />
        </List>
      </Card>
    );
  }
}

ResourcePage.propTypes = {
  router: PropTypes.any,
};
