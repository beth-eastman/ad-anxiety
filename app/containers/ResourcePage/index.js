/**
 * Contains video contents of the application.
 */

import React from 'react';
import { Card, CardHeader, List, ListItem } from 'material-ui';
import { browserHistory } from 'react-router';

export default class ResourcePage extends React.Component {

  componentWillMount() {
    this.props.setPageTitle("Resources"); // eslint-disable-line
  }

  render() {
    return (
      <Card>
        <CardHeader
          title="Resources"
        />
        <List>
          <ListItem
            primaryText="Links & Resources"
            onTouchTap={() => {
              browserHistory.push('/resources/page1');
            }}
          />
          <ListItem
            primaryText="Forum"
            onTouchTap={() => {
              browserHistory.push('/resources/page2');
            }}
          />
          <ListItem
            primaryText="Facts"
            onTouchTap={() => {
              browserHistory.push('/resources/page3');
            }}
          />
          <ListItem
            primaryText="Articles"
            onTouchTap={() => {
              browserHistory.push('/resources/page4');
            }}
          />
        </List>
      </Card>
    );
  }
}
