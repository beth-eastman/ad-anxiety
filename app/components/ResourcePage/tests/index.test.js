/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import ResourcePage from '../index';

it('renders ResourcePage', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <ResourcePage params={{ factId: 0 }} setPageTitle={function () {}} changeIcon={function () {}} />
    </MuiThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
