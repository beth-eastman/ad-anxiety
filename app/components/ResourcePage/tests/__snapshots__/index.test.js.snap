exports[`test renders ResourcePage 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "8px 0px 8px 0px",
        }
      }>
      <div>
        <span
          disabled={undefined}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          open={null}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "backgroundColor": null,
              "border": 10,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "fontWeight": "inherit",
              "lineHeight": "16px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={0}>
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}>
            <div
              style={
                Object {
                  "marginLeft": 0,
                  "muiPrepared": true,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "position": "relative",
                }
              }>
              <div
                style={
                  Object {
                    "muiPrepared": true,
                  }
                }>
                Links
              </div>
            </div>
          </div>
        </span>
      </div>
      <div>
        <span
          disabled={undefined}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          open={null}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "backgroundColor": null,
              "border": 10,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "fontWeight": "inherit",
              "lineHeight": "16px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={0}>
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}>
            <div
              style={
                Object {
                  "marginLeft": 0,
                  "muiPrepared": true,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "position": "relative",
                }
              }>
              <div
                style={
                  Object {
                    "muiPrepared": true,
                  }
                }>
                Books
              </div>
            </div>
          </div>
        </span>
      </div>
      <div>
        <span
          disabled={undefined}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          open={null}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "backgroundColor": null,
              "border": 10,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "fontWeight": "inherit",
              "lineHeight": "16px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={0}>
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}>
            <div
              style={
                Object {
                  "marginLeft": 0,
                  "muiPrepared": true,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "position": "relative",
                }
              }>
              <div
                style={
                  Object {
                    "muiPrepared": true,
                  }
                }>
                Articles
              </div>
            </div>
          </div>
        </span>
      </div>
      <div>
        <span
          disabled={undefined}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          open={null}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "backgroundColor": null,
              "border": 10,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "fontWeight": "inherit",
              "lineHeight": "16px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={0}>
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}>
            <div
              style={
                Object {
                  "marginLeft": 0,
                  "muiPrepared": true,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "position": "relative",
                }
              }>
              <button
                disabled={false}
                href={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOut={[Function]}
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "background": "none",
                    "border": 10,
                    "boxSizing": "border-box",
                    "cursor": "pointer",
                    "display": "block",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": 0,
                    "fontWeight": "inherit",
                    "height": 48,
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "overflow": "visible",
                    "padding": 12,
                    "position": "absolute",
                    "right": 4,
                    "textDecoration": "none",
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "verticalAlign": null,
                    "width": 48,
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                type="button">
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}>
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "display": "inline-block",
                        "fill": "currentColor",
                        "height": 24,
                        "muiPrepared": true,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24">
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                  </svg>
                </div>
              </button>
              <div
                style={
                  Object {
                    "muiPrepared": true,
                  }
                }>
                Facts
              </div>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
